tests/lang/core.pk

```diff
@@ -30,6 +30,15 @@ def fn(p1, p2, p3) end
 assert(fn.name == "fn")
 assert(fn.arity == 3)
 
+## String functions
+assert(str_sub('c programming', 2, 11) == 'programming')
+assert(str_sub('foobarbaz', 2, 3) == 'oba')
+assert(str_sub('abcdef', 1, 2) == 'bc')
+assert(str_sub('I am a boy', 0, 4) == 'I am')
+assert(str_sub('programming', 2, 0) == '')
+assert(str_sub('foobar', 5, 0) == '')
+assert(str_sub('foobar', 0, 6) == 'foobar')
+
```

Comment:
add `assert(str_sub('', 0, 0) == '')` after removing the string length validation I've mentioned above.